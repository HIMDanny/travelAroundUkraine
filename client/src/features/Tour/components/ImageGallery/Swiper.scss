.swiper {
  width: 100%;
  height: 500px;
  max-width: 1200px;
}

.swiper-slide {
  text-align: center;
  font-size: 18px;
  background: #fff;
  width: auto !important;
  display: flex;
  justify-content: center;
  align-items: center;
}

.img-inside {
  margin: 20px 0;
}

.swiper-slide img {
  display: block;
  width: 100%;
  height: 100%;
  object-fit: contain;
}

.c {
  position: relative;
  max-width: 1200px;
  margin: 0 auto;
}

.thumbsSwiper {
  margin-top: 10px;
}

.thumbsSwiper .swiper-slide {
  width: 25%;
  height: 100%;
  opacity: 0.7;
}

.thumbsSwiper .swiper-slide-thumb-active {
  opacity: 1;
}

.react-simple-image-viewer__modal {
  z-index: 2000 !important;
}

.stop-scrolling {
  height: 100%;
  overflow: hidden;
}

@media screen and (max-width: 1220px) {
  .c {
    margin: 0 20px 20px;
  }
}

@media screen and (max-width: 820px) {
  .c {
    margin: 70px 20px 20px;
  }

  #n,
  #p {
    display: none;
  }
}

@media screen and (max-width: 500px) {
  .mySwiper2 {
    height: 300px;
  }

  .mySwiper2 .swiper-wrapper .swiper-slide {
    text-align: center;
    font-size: 18px;
    background: #fff;
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    object-fit: contain;
  }
}
