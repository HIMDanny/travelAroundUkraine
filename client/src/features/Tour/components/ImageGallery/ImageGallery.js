/* eslint-disable react/prop-types */
/* eslint-disable import/no-unresolved */
import React, { useState, useCallback } from 'react';
import ImageViewer from 'react-simple-image-viewer';
import { Swiper, SwiperSlide } from 'swiper/react';

import 'swiper/swiper-bundle.css';
import './Swiper.scss';

import { Autoplay, FreeMode, Mousewheel, Scrollbar, Zoom, Thumbs, Navigation } from 'swiper';

import { SliderButton } from '../../../../components';

export default function ImageGallery({ imageUrls }) {
  const [currentImage, setCurrentImage] = useState(0);
  const [isViewerOpen, setIsViewerOpen] = useState(false);
  const [thumbsSwiper, setThumbsSwiper] = useState(null);

  // const image = [
  //   'http://placeimg.com/1200/800/nature',
  //   'http://placeimg.com/800/1200/nature',
  //   'http://placeimg.com/800/1200/nature',
  //   'http://placeimg.com/1920/1080/nature',
  //   'http://placeimg.com/1500/500/nature',
  //   'http://placeimg.com/1500/500/nature',
  //   'http://placeimg.com/800/1200/nature',
  //   'http://placeimg.com/1500/500/nature',
  //   'http://placeimg.com/1200/800/nature',
  //   'http://placeimg.com/1200/800/nature',
  //   'http://placeimg.com/1200/800/nature',
  // ];

  function disableScroll() {
    document.body.classList.add('stop-scrolling');
  }
  function enableScroll() {
    document.body.classList.remove('stop-scrolling');
  }

  const openImageViewer = useCallback((index) => {
    setCurrentImage(index);
    setIsViewerOpen(true);
    disableScroll();
  }, []);

  const closeImageViewer = () => {
    setCurrentImage(0);
    setIsViewerOpen(false);
    enableScroll();
  };

  // to return a preloader in the future
  if (!imageUrls) return null;

  return (
    <div className="c">
      <SliderButton id="p" position={{ top: '40%', left: '-30px', rotate: '180deg' }} />
      <SliderButton id="n" position={{ top: '40%', right: '-30px' }} />
      <Swiper
        spaceBetween={5}
        zoom
        loop
        longSwipes
        autoplay={{
          delay: 2500,
          disableOnInteraction: true,
          pauseOnMouseEnter: true,
        }}
        thumbs={{ swiper: thumbsSwiper }}
        watchSlidesProgress
        navigation={{
          nextEl: '#n',
          prevEl: '#p',
        }}
        mousewheel
        modules={[Autoplay, Mousewheel, FreeMode, Scrollbar, Zoom, Navigation, Thumbs]}
        slidesPerView="auto"
        className="mySwiper2"
      >
        {imageUrls.map((src, index) => (
          // eslint-disable-next-line react/no-array-index-key
          <SwiperSlide key={`${src + index}`}>
            {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-noninteractive-element-interactions */}
            <img src={src} alt="" onClick={() => openImageViewer(index)} />
          </SwiperSlide>
        ))}
      </Swiper>
      <Swiper
        style={{
          height: '70px',
          width: 'auto',
        }}
        onSwiper={setThumbsSwiper}
        spaceBetween={10}
        slidesPerView="auto"
        watchSlidesProgress
        modules={[Navigation, Thumbs]}
        className="thumbsSwiper"
      >
        {imageUrls.map((src, i) => (
          <SwiperSlide key={`${src + i}`}>
            <img src={src} alt="" />
          </SwiperSlide>
        ))}
      </Swiper>
      {isViewerOpen && (
        <ImageViewer
          src={imageUrls}
          currentIndex={currentImage}
          onClose={closeImageViewer}
          disableScroll
          backgroundStyle={{
            backgroundColor: 'rgba(0,0,0,0.9)',
            zIndex: 10,
          }}
          closeOnClickOutside
        />
      )}
    </div>
  );
}
