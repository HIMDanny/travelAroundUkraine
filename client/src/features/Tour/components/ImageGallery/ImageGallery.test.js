import { render } from '@testing-library/react';
import ImageGallery from './ImageGallery';

jest.mock('../../../../components/SliderButton/SliderButton', () => () => <button>test button</button>);

describe('Swiper has to render', () => {
  test('swiper renders', () => {
    const images = ['test1', 'test2', 'test3'];

    const { asFragment } = render(<ImageGallery imageUrls={images} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
