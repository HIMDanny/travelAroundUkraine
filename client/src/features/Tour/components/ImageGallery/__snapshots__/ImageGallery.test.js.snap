// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swiper has to render swiper renders 1`] = `
<DocumentFragment>
  <div
    class="c"
  >
    <button>
      test button
    </button>
    <button>
      test button
    </button>
    <div
      class="swiper swiper-initialized swiper-horizontal swiper-pointer-events swiper-watch-progress mySwiper2"
    >
      <div
        class="swiper-wrapper"
        style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
      >
        <div
          class="swiper-slide swiper-slide-duplicate"
          data-swiper-slide-index="0"
        >
          <img
            alt=""
            src="test1"
          />
        </div>
        <div
          class="swiper-slide swiper-slide-duplicate"
          data-swiper-slide-index="1"
        >
          <img
            alt=""
            src="test2"
          />
        </div>
        <div
          class="swiper-slide swiper-slide-duplicate"
          data-swiper-slide-index="2"
        >
          <img
            alt=""
            src="test3"
          />
        </div>
        <div
          class="swiper-slide"
          data-swiper-slide-index="0"
        >
          <img
            alt=""
            src="test1"
          />
        </div>
        <div
          class="swiper-slide"
          data-swiper-slide-index="1"
        >
          <img
            alt=""
            src="test2"
          />
        </div>
        <div
          class="swiper-slide"
          data-swiper-slide-index="2"
        >
          <img
            alt=""
            src="test3"
          />
        </div>
        <div
          class="swiper-slide swiper-slide-duplicate"
          data-swiper-slide-index="0"
        >
          <img
            alt=""
            src="test1"
          />
        </div>
        <div
          class="swiper-slide swiper-slide-duplicate"
          data-swiper-slide-index="1"
        >
          <img
            alt=""
            src="test2"
          />
        </div>
        <div
          class="swiper-slide swiper-slide-duplicate"
          data-swiper-slide-index="2"
        >
          <img
            alt=""
            src="test3"
          />
        </div>
      </div>
    </div>
    <div
      class="swiper swiper-initialized swiper-horizontal swiper-pointer-events swiper-watch-progress thumbsSwiper swiper-thumbs"
      style="height: 70px; width: auto;"
    >
      <div
        class="swiper-wrapper"
        style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
      >
        <div
          class="swiper-slide"
        >
          <img
            alt=""
            src="test1"
          />
        </div>
        <div
          class="swiper-slide"
        >
          <img
            alt=""
            src="test2"
          />
        </div>
        <div
          class="swiper-slide"
        >
          <img
            alt=""
            src="test3"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
